### ISSUE TITLE
cpp files for mac compile on linux as well

### ISSUE THREAD

**Issue by underdoeg** [ofZach/ofxNativeGui#3]
Hey, I have a partially working linux version of your addon. Dropdown selector doesn't work yet.

https://github.com/underdoeg/ofxNativeGui

Problem is that the way the makefile works on linux, it automatically tries to compile the mac related cpp files as well. Maybe you could rename them to .mm? This won't fix the windows specific files though...

**Comment by ofZach** [comment]
would one simple option be to throw huge ifdef around all the apple code?

```
 #ifdef __APPLE__

 #endif
```

we do that, for example in ofxOsc, since there's windows and posix related code -- we basically have whole files you only compile if you are windows and vice versa.

**Comment by underdoeg** [comment]
Yes, of course that works as well. Not a great solution but ok...

Unrelated, but let's look at this issue as a general linux issue :) You say 

> Some other things I do are make the int sliders float compatible, etc.

Did you also change something with the combo box? Because using it freezes the gui window and I am not sure why...
